src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1588,6 +1588,9 @@ void PeerReplayer::run(SnapshotReplayerThread *replayer) {
 	  r = sync_perms(*dir_root);
 	  if (r < 0) {
 	    _inc_failed_count(*dir_root);
+	    if (m_perf_counters) {
```

Comment:
Suggest we do this in a single place in this source here thereby removing the one ins sync_snaps() and handling the failure cases in run().

Reply:
@vshankar Fixed. Please check.